fix strategy hashing arg mismatch

zpcore opened the pull request:
Reland https://github.com/pytorch/pytorch/pull/159289.


cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta

XilunWu reviewed (approved):
reland of [159289](https://github.com/pytorch/pytorch/pull/159289)

zpcore commented:
@pytorchbot merge -f "just just merge"

yangw-dev commented:
@pytorchbot revert -m "failed the internal tests test_get_bwd_hook
        self.assertTrue(torch.equal(output * 2, input_tensor.grad)) AssertionError: False is not true" -c ghfirst

yangw-dev commented:
@pytorchbot revert -m -m "failed the internal tests test_get_bwd_hook self.assertTrue(torch.equal(output * 2, input_tensor.grad)) AssertionError: False is not true" -c ghfirst

yangw-dev commented:
@pytorchbot revert -m -m "failed the internal tests test_get_bwd_hook self.assertTrue(torch.equal(output *2,input_tensor.grad)) AssertionError: False is not true" -c ghfirst

yangw-dev commented:
@pytorchbot revert -m "failed the internal tests test_get_bwd_hook (torch.equal(output * 2, input_tensor.grad))" -c ghfirst

zpcore commented:
@pytorchbot rebase

zpcore committed:
fix strategy hashing arg mismatch (#159506)

Summary:
Reland https://github.com/pytorch/pytorch/pull/159289.


cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta


Reviewed By: XilunWu

Differential Revision: D79404462

Pulled By: zpcore

zpcore commented:
@pytorchbot merge -f "just just merge"
